include/bluetooth/bluetooth.h

```diff
@@ -704,6 +704,14 @@ struct bt_le_per_adv_param {
 						BT_GAP_ADV_FAST_INT_MAX_2, \
 						NULL)
 
+/** Scannable extended advertising with @ref BT_LE_ADV_OPT_USE_NAME */
+#define BT_LE_EXT_ADV_SCAN_NAME BT_LE_ADV_PARAM(BT_LE_ADV_OPT_EXT_ADV | \
+						BT_LE_ADV_OPT_SCANNABLE | \
+						BT_LE_ADV_OPT_USE_NAME, \
```

Comment:
I was wondering why the other default parameters didn't have `BT_LE_ADV_OPT_SCANNABLE` when they are putting the device name in the scan response. From what I can tell from the stack code, `BT_LE_ADV_OPT_SCANNABLE` is implicitly set when `BT_LE_ADV_OPT_USE_NAME` is set, or if any scan data was provided in which doesn't seem to be documented at all. It even appears that if any `sd` is provided in `bt_le_adv_start` then `BT_LE_ADV_OPT_SCANNABLE` is also implicitly enabled. 

Is this by design and intention @joerchan ?

Reply:
That is correct w.r.t. `sd` not being there anymore. However, the expression that adds scan data uses a boolean based on `sd || (param->options & BT_LE_ADV_OPT_USE_NAME)`, so even with the extended advertising API, if `BT_LE_ADV_OPT_USE_NAME` is set, then it is scannable. 

I actually see that it is not taking the `BT_LE_ADV_OPT_FORCE_NAME_IN_AD` option into account for that, so I have also created this issue: https://github.com/zephyrproject-rtos/zephyr/issues/37462

Reply:
> the expression that adds scan data uses a boolean based on sd || (param->options & BT_LE_ADV_OPT_USE_NAME)

Correct me, that piece of code path is only applicable for legacy advertising... `bt_le_ext_adv_create` always sets `false` to `has_scan_data` parameter when calling `le_ext_adv_param_set`

Reply:
> Is this by design and intention @joerchan ?

Yes, ADV_OPT_SCANNABLE was added together with the extended advertising API, and it is not required for the legacy API in order to make the advertiser scannable. Setting the SD data (or OPT_NAME) makes it scannable implicitly. This maintains the old behavior and application did not have to update.

Reply:
@cvinayak The check also appears in `bt_le_adv_start_ext`:
```
	err = le_ext_adv_param_set(adv, param, sd ||
				   (param->options & BT_LE_ADV_OPT_USE_NAME));
```

@joerchan Is this perhaps something that should be made clear in the API description?